subworkflows/local/bam_variant_calling_haplotypecaller/main.nf

```diff
@@ -1,168 +1,120 @@
+    haplotypecaller_tbi = GATK4_HAPLOTYPECALLER.out.tbi.branch{
+        // Use meta.num_intervals to asses number of intervals
             intervals:    it[0].num_intervals > 1
             no_intervals: it[0].num_intervals <= 1
         }
 
-    haplotypecaller_bam_branch = GATK4_HAPLOTYPECALLER.out.bam.branch{
+    // Figuring out if there is one or more bam(s) from the same sample
+    haplotypecaller_bam = GATK4_HAPLOTYPECALLER.out.bam.branch{
+        // Use meta.num_intervals to asses number of intervals
             intervals:    it[0].num_intervals > 1
             no_intervals: it[0].num_intervals <= 1
         }
 
-    if (params.joint_germline) {
-        // merge vcf and tbis
-        genotype_gvcf_to_call = Channel.empty().mix(
-                GATK4_HAPLOTYPECALLER.out.vcf
-                .join(GATK4_HAPLOTYPECALLER.out.tbi)
-                .join(cram)
-                .map{ meta, vcf, tbi, cram, crai, intervals, dragstr_model ->
-                    [ meta, vcf, tbi, intervals ]
-                })
+    // Only when using intervals
+    MERGE_HAPLOTYPECALLER(haplotypecaller_vcf.intervals
+        .map{ meta, vcf -> [ groupKey(meta, meta.num_intervals), vcf ] }
+        .groupTuple(),
+        dict.map{ it -> [ [ id:'dict' ], it ] })
+
+    haplotypecaller_vcf = Channel.empty().mix(
+            MERGE_HAPLOTYPECALLER.out.vcf,
+            haplotypecaller_vcf.no_intervals)
+
+    haplotypecaller_tbi = Channel.empty().mix(
+            MERGE_HAPLOTYPECALLER.out.tbi,
+            haplotypecaller_tbi.no_intervals)
+
+    // BAM output
+    BAM_MERGE_INDEX_SAMTOOLS(haplotypecaller_bam.intervals
+        .map{ meta, bam -> [ groupKey(meta, meta.num_intervals), bam ] }
+        .groupTuple()
+        .mix(haplotypecaller_bam.no_intervals))
 
-        // make channels from labels
-        dbsnp_vqsr        = params.dbsnp_vqsr        ? Channel.value(params.dbsnp_vqsr)        : Channel.empty()
-        known_indels_vqsr = params.known_indels_vqsr ? Channel.value(params.known_indels_vqsr) : Channel.empty()
-        known_snps_vqsr   = params.known_snps_vqsr   ? Channel.value(params.known_snps_vqsr)   : Channel.empty()
+    realigned_bam = BAM_MERGE_INDEX_SAMTOOLS.out.bam_bai
 
+    if (!skip_haplotypecaller_filter) {
 
-        BAM_JOINT_CALLING_GERMLINE_GATK(
-            genotype_gvcf_to_call,
+        VCF_VARIANT_FILTERING_GATK(
+            haplotypecaller_vcf.join(haplotypecaller_tbi),
             fasta,
             fasta_fai,
             dict,
-            dbsnp,
-            dbsnp_tbi,
-            dbsnp_vqsr,
-            known_sites_indels,
-            known_sites_indels_tbi,
-            known_indels_vqsr,
-            known_sites_snps,
-            known_sites_snps_tbi,
-            known_snps_vqsr)
-
-        vcf = BAM_JOINT_CALLING_GERMLINE_GATK.out.genotype_vcf
-        versions = versions.mix(BAM_JOINT_CALLING_GERMLINE_GATK.out.versions)
-
-    } else {
```

Comment:
I don't think the filtering here is in the scope of this subwf (but correct me if I'm wrong :grin: )

Reply:
one subworkflow per process? :D I think in the scope of singlesample gatk vc it would be, since this is what is recommended by the GATK best-practices, ut we can argue about this some more, when actually moving this over

Reply:
Allright fine by me :p